### DISCUSSION TITLE
Disable include by query string

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#122] [Q&A]
Hello everybody,

I am just wondering, whether it's possible to disable this behavior: https://spatie.be/docs/laravel-data/v1/as-a-resource/lazy-properties#including-by-query-string .

Reason: I got really large nested objects and I need to prevent people to set queries, that create a really large call stack.

**Comment by rubenvanassche** [comment]
Hi @ahoiroman,

Not at the moment but in v2 we change the behavior into an opt in behavior where you should define which properties can be included by query. By default all properties can be included but returning an empty array here disables the feature: https://github.com/spatie/laravel-data/blob/51d37055cb3769a37424560ef64571576f31f727/src/Concerns/ResponsableData.php#L24

Hope we can release v2 really soon!

**Reply by ahoiroman** [reply]
Thanks. So I am really excited for v2 :-) Thanks for your work!
